elasticsearch-cloud-mesos/src/main/java/org/elasticsearch/cloud/mesos/MesosStateServiceMesosDns.java

```diff
@@ -56,14 +60,15 @@ protected void doClose() throws ElasticsearchException {
 
                 while (enumeration.hasMore()) {
                     SrvRecord record = new SrvRecord((String) enumeration.next());
-
                     nodes.add(record.getHostAndPort());
                 }
             }
         } catch (NamingException e) {
-            throw new RuntimeException("Failed to resolve hostname", e);
+            logger.debug("Failed to resolve hostname", e.getCause());
```

Comment:
getting a cause is a little bit dangerous as you're actually hiding an exception